File: src/main/scala/org/bakkes/game/controller/state/battle/contestent/RandyRandom.java
package org.bakkes.game.controller.state.battle.contestent;

import com.google.inject.Inject;
import org.bakkes.game.model.battle.move.IMove;
import scala.collection.Seq;

import java.util.Random;


public class RandyRandom extends AI{

	@Inject Random random;
	@Override
	protected IMove getMove() {
		final Seq<IMove> own = getOwnPokemon().getMoves();
		return own.take(random.nextInt(own.length())).last();
	}

}

File: src/main/scala/org/bakkes/game/controller/state/battle/contestent/RandyRandom.scala
package org.bakkes.game.controller.state.battle.contestent

import java.util.Random

import com.google.inject.Inject
import org.bakkes.game.model.battle.move.IMove
import org.newdawn.slick.util.Log

class RandyRandom @Inject() (
	val random: Random
)extends AI {
	protected def getMove: IMove = {
		val own = getOwnPokemon.getMoves
		val choice = random.nextInt(own.length)
		Log.debug("ai choice " + choice + " from: " + getOwnPokemon.getMoves)
		return own.apply(choice)
	}
}